### PR TITLE
Referenced_filenames should be returned by API function

### PR DESCRIPTION
**Pull request by akugarg**
Signed-off-by: akugarg <akanksha.garg2k@gmail.com>

Check https://github.com/nexB/scancode-toolkit/pull/2616#discussion_r684384317 for more details

### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁

### CODE REVIEW

File: src/licensedcode/models.py [hunk 1]
```diff
@@ -877,6 +877,15 @@ def has_unknown(self):
         # TODO: consider using the license_expression_object and the is_unknown
         # license flag instead
         return self.license_expression and 'unknown' in self.license_expression
+    
+    @property
+    def reference_files(self):
```
Comment (AyanSinhaMahapatra) [inline src/licensedcode/models.py]: Is this needed? We could directly add `match.rule.referenced_filenames` [here](https://github.com/nexB/scancode-toolkit/pull/2632/files#diff-9b8c9108cb6f529dc48ef0aba7a714c7e2b53feac33c7dc91eac423685a194b7R269). @pombredanne what do you think?

Comment (akugarg) [reply src/licensedcode/models.py]: oh yeah! you are right ! I'll change it

**Comment by akugarg** [comment]
@AyanSinhaMahapatra  All green! 
@pombredanne plz have a look!

File: tests/summarycode/data/full_summary/summary_key_files.expected.json [hunk 2]
```diff
@@ -2446,7 +2481,7 @@
             "license_expression": "zlib",
             "licenses": [
               "zlib"
-            ],
+            ],"referenced_filenames": [],
```
Comment (AyanSinhaMahapatra) [inline tests/summarycode/data/full_summary/summary_key_files.expected.json]: Minor nit, should be in a new line.

**Review by AyanSinhaMahapatra — APPROVED** [review]
Thanks! Looks ready.

**Review by pombredanne — APPROVED** [review]
Perfect! 
LGTM.... Please also add a CHANGELOG.rst entry for this and the feature of follwing these references (this can be in another PR)

**Pull request merged**